src/Features/LanguageServer/Protocol/LanguageServerTarget.cs

```diff
@@ -54,28 +52,73 @@ internal class LanguageServerTarget : ILanguageServerTarget
             string? clientName,
             WellKnownLspServerKinds serverKind)
         {
-            GlobalOptions = globalOptions;
-            RequestDispatcher = requestDispatcherFactory.CreateRequestDispatcher(supportedLanguages, serverKind);
+            _requestDispatcher = requestDispatcherFactory.CreateRequestDispatcher(serverKind);
 
             _capabilitiesProvider = capabilitiesProvider;
-            WorkspaceRegistrationService = workspaceRegistrationService;
-            Logger = logger;
+            _logger = logger;
 
-            JsonRpc = jsonRpc;
-            JsonRpc.AddLocalRpcTarget(this);
-            JsonRpc.Disconnected += JsonRpc_Disconnected;
+            _jsonRpc = jsonRpc;
+            _jsonRpc.AddLocalRpcTarget(this);
+            _jsonRpc.Disconnected += JsonRpc_Disconnected;
 
-            Listener = listenerProvider.GetListener(FeatureAttribute.LanguageServer);
-            ClientName = clientName;
+            _listener = listenerProvider.GetListener(FeatureAttribute.LanguageServer);
+            _clientName = clientName;
 
-            Queue = new RequestExecutionQueue(
+            _queue = new RequestExecutionQueue(
                 logger,
                 workspaceRegistrationService,
                 lspMiscellaneousFilesWorkspace,
                 globalOptions,
                 supportedLanguages,
                 serverKind);
-            Queue.RequestServerShutdown += RequestExecutionQueue_Errored;
+            _queue.RequestServerShutdown += RequestExecutionQueue_Errored;
+
+            foreach (var metadata in _requestDispatcher.GetRegisteredMethods())
+            {
+                // Instead of concretely defining methods for each LSP method, we instead dynamically construct
+                // the generic method info from the exported handler types.  This allows us to define multiple handlers for the same method
+                // but different type parameters.  This is a key functionality to support TS external access as we do not want to couple
+                // our LSP protocol version dll to theirs.
+                //
+                // We also do not use the StreamJsonRpc support for JToken as the rpc method parameters because we want
+                // StreamJsonRpc to do the deserialization to handle streaming requests using IProgress<T>.
+                var delegatingEntryPoint = new DelegatingEntryPoint(metadata.MethodName, this);
+
+                var entryPointMethod = delegatingEntryPoint.GetType().GetMethod(nameof(DelegatingEntryPoint.EntryPointAsync));
```

Comment:
Nit: Reflection should cache this, but still simpler to do this outside of the loop, with `typeof(DelegatingEntryPoint)`